exporter/exporterhelper/common.go

```diff
@@ -47,7 +46,7 @@ type request interface {
 	setContext(context.Context)
 	export(ctx context.Context) error
 	// Returns a new request that contains the items left to be sent.
-	onPartialError(consumererror.PartialError) request
+	onPartialError(error) request
```

Comment:
I would change this to "onError" and implement the behavior to return the same request if it is not the signal error. Then we don't need to call isPartial anymore.

Reply:
Done.  This results in a bit more error checking at each implementation, but since each implementation has different things it wants to deal with that doesn't seem unreasonable.